pkg/providers/vsphere/virtualmachine/backup.go

```diff
@@ -115,12 +177,48 @@ func BackupVirtualMachine(opts BackupVirtualMachineOptions) error {
 			return err
 		}
 
+		// Set the VirtualMachine's backup version annotation once reconfigure succeeds.
+		if pkgcfg.FromContext(opts.VMCtx).Features.VMIncrementalRestore {
+			setLastBackupVersionAnnotation(opts.VMCtx.VM, opts.BackupVersion)
+		}
+
 		opts.VMCtx.Logger.Info("Successfully updated VM ExtraConfig with latest backup data")
 	}
 
 	return nil
 }
 
+// isBackupNeeded checks if a backup can be performed. The method returns true when
+// the backup version annotation matches (or) is older (lesser) than the backup version in extraConfig.
+// It returns false when the backup version annotation is newer (greater) than the backup extraConfig version which indicates a vendor
+// triggered restore in progress.
+func isBackupNeeded(opts BackupVirtualMachineOptions, backupVersionEcVal, backupVersionAnnotation string) (bool, error) {
```

Comment:
I would recommend renaming this to something like `shouldPerformBackup` `canPerformBackup`, or `canBackup` because even if a backup is needed (because the config has changed on Supervisor), this method can return false.